tracer/src/Datadog.Trace/Debugger/LiveDebugger.cs

```diff
@@ -106,7 +106,6 @@ public async Task InitializeAsync()
                 DebuggerSnapshotSerializer.SetConfig(_settings);
                 Product.ConfigChanged += (sender, args) => AcceptConfiguration(args);
                 AppDomain.CurrentDomain.AssemblyLoad += (sender, args) => CheckUnboundProbes();
-                AppDomain.CurrentDomain.DomainUnload += (sender, args) => _lineProbeResolver.OnDomainUnloaded();
```

Comment:
So, this is a behaviour change. Is that _definitely_ correct? 🤔

Reply:
I took a look and AFAICT this doesn't change behavior in any meaningful way - what `LineProbeResolver.OnDomainUnloaded` does in clear out a bunch of managed objects, and those managed objects can only be inserted after `StartAsync` is called.

Having said that, the fact that all it's doing is clearing out some managed objects means that this code is not really necessary and I think we can safely just delete it, as those managed objects will die automatically when the `AppDomain` unloads. `LineProbeResolver.OnDomainUnloaded` is probably a left-over from a time when the `LineProbeResolver` was listening to events that had to be unregistered, in order to avoid interfering with the `AppDomain` shutdown.